### PR TITLE
remove linux version of oak functions

### LINKED ISSUE #3738: Remove linux variant for of Oak Functions
We should consider dropping the linux variant, and doing everything with the KVM variant only (e.g. in tests and examples). I don't see much value in keeping both at this point, since we are already set on KVM.

The main advantage of the linux variant would be that it can run on machines without KVM, and perhaps that it's marginally faster, but booting the Oak Restricted Kernel seems fast enough that we can use it for everything.

WDYT @andrisaar ?

### PR DESCRIPTION
**Pull request by jul-sh**
fixes #3738

### CODE REVIEW

File: oak_functions_launcher/tests/integration_test.rs [hunk 1]
```diff
@@ -159,185 +149,3 @@ async fn test_launcher_weather_lookup_virtual() {
...
-    launched_instance
-        .kill()
-        .await
-        .expect("Failed to stop launcher");
-}
-
-#[ignore = "too expensive"]
-#[tokio::test]
-async fn test_load_two_gib_lookup_data() {
-    let oak_functions_linux_fd_bin_path =
```
Comment (jul-sh) [inline oak_functions_launcher/tests/integration_test.rs:313]: this effectively removes the test for large amounts of lookup data. Lookup data functionality remains tested. Per discussion in https://github.com/project-oak/oak/issues/3738, this should be fine

**Review by conradgrobler — APPROVED** [review]

File: oak_functions_launcher/tests/integration_test.rs (hunk 1 above)
Comment (mariaschett) [reply oak_functions_launcher/tests/integration_test.rs:313]: Is there no way we can keep the test---and the one above `test_load_large_lookup_data`? I am not sure I have all the context for the PR, but I am not sure we have tests for this elsewhere.

**Review by ipetr0v — APPROVED** [review]

File: oak_functions_launcher/tests/integration_test.rs (hunk 1 above)
Comment (jul-sh) [reply oak_functions_launcher/tests/integration_test.rs:313]: I'll look into it!

Comment (jul-sh) [reply oak_functions_launcher/tests/integration_test.rs]: @mariaschett I updated the two tests to work with the KVM version and added them back to this PR. Could you have another look? :)

**Review by mariaschett — APPROVED** [review]
Thank you for refactoring the tests!

**Pull request merged**